gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -206,4 +206,53 @@ var fluid = fluid || require("infusion"),
         });
         return togo;
     };
+
+    /*
+     * Converts application specific settings into common terms, keyed by application ids. Given a
+     * set of preferences, for each application block in those preferences, the settings for which
+     * inverse transformations exist will be transformed into common terms. All the common terms
+     * that have successfully been inferred from an application will be written to a block keyed by
+     * that applications ID.
+     *
+     * @preferences (Object) - An NP set (including contexts, etc)
+     *
+     * @return (Object) - Object where keys are solution IDs and entries are common term
+     *      preferences which have been transformed from application specific settings for that
+     *      soluion
+     */
+    gpii.matchMakerFramework.utils.inferCommonTerms = function (preferences, fullSolutionsRegistry) {
```

Comment:
I think this is still required here as we need to check all the different OS entries for inverse transformations - but it's modelled as a datasource as requested in the matchmaker code. It is just passed to the inferCommonTerms utility as the full object